### PR TITLE
homework database week1

### PR DESCRIPTION
**Pull request by tiago86g**

### CODE REVIEW

File: databases/week1/db_w1.sql [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+select count(id)
+from task;
+
+-- 2 Find out how many tasks in the task table do not have a valid due date
+select count(id)
+from task
+where due_date IS null;
+
+
+-- 3 Find all the tasks that are marked as done (what should I do with null response?)
```
Comment (babakbandpey) [inline databases/week1/db_w1.sql]: Would you please solve this by the JOIN operator?
Thank you

File: databases/week1/db_w1.sql [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+select id
+from task
+where status_id = 3;
+
+-- 4 Find all the tasks that are not marked as done
```
Comment (babakbandpey) [inline databases/week1/db_w1.sql]: This one needs to be solved by JOIN too

File: databases/week1/db_w1.sql [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+-- 7 Get the title and due date of all tasks where the title or description contains database
+select title, due_date
+from task
+where title is null or due_date is null;
+
+
+-- 8 Get the title and status (as text) of all tasks
+select title, name
```
Comment (moonflare) [inline databases/week1/db_w1.sql]: Valid query, but it's good practice to use aliases to make sure the results are as clear as possible.
Also, when using `JOIN`s it's good practice to use `<tableName>.<columnName>` so you or other people visiting the code later know where your columns come from.
```suggestion
select task.title, status.name as status
```

File: databases/week1/db_w1.sql [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+from task join status 
+on status.id = task.status_id;
+
+
+-- 9 Get the name of each status, along with a count of how many tasks have that status
+select count(*), name
+from task join status on task.status_id = status.id
```
Comment (moonflare) [inline databases/week1/db_w1.sql]: Even though this query works for the set of data we have, what would happen if we add an extra status named `Postponed`? Also, how could we solve the problem?

File: databases/week1/db_w1.sql [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+order by created desc;
+
+-- 6 Get the single most recently created task
+select id, created
+from task
+order by created desc
+limit 6;
+
+
+-- 7 Get the title and due date of all tasks where the title or description contains database
```
[the remark this answers is not in the document]
Comment (tiago86g) [reply databases/week1/db_w1.sql:35]: Thanks for the feedback I’ve  improved it.

File: databases/week1/db_w1.sql (hunk 2 above)
Comment (tiago86g) [reply databases/week1/db_w1.sql:17]: Thanks Babak I’ve improved  it now

File: databases/week1/db_w1.sql (hunk 1 above)
Comment (tiago86g) [reply databases/week1/db_w1.sql:11]: I have improved theses exercises. Thanks for your feedback :)

File: databases/week1/db_w1.sql (hunk 4 above)
Comment (tiago86g) [reply databases/week1/db_w1.sql:49]: Thanks for the feedback. I solved theses issues.

**Pull request closed without merging**